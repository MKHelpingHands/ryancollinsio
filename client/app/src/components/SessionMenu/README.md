## SessionMenu Component
A component that ...

### Example

```js
<SessionMenu />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
