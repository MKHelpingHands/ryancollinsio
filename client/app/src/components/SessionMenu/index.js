import React, { PropTypes } from 'react';
import Menu from 'grommet-udacity/components/Menu';
import Anchor from 'grommet-udacity/components/Anchor';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Box';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
const noAvatar = 'http://bit.ly/2dqCGdd';

const NoUserMenu = ({ pathname }) => (
  <Menu
    direction="row"
    align="center"
    inline
    resposive={false}
    className={styles.rightMenu}
  >
    <Anchor
      className={pathname === '/login' ? 'navLink active' : 'navLink'}
      href="/login"
    >
      Log In
    </Anchor>
    <Anchor
      className={pathname === '/signup' ? 'navLink active' : 'navLink'}
      href="/signup"
    >
      Sign Up
    </Anchor>
  </Menu>
);

const SessionIcon = ({
  user,
}) => (
  <Box
    responsive={false}
    direction="row"
    justify="center"
  >
    <img
      src={user.avatar || noAvatar}
      className={styles.userAvatar}
    />
    <Heading tag="h4" className={styles.profileName}>
      {user.name}
    </Heading>
  </Box>
);

const SessionMenu = ({
  onLogout,
  user,
  pathname,
  tabIndex,
}) => (
  <span>
    {(() => { // eslint-disable-line
      if (user) {
        switch (user.role) {
          case 'user':
            return (
              <Menu
                tabIndex={tabIndex}
                icon={<SessionIcon user={user} />}
                inline={false}
                dropAlign={{ top: 'top', right: 'right' }}
                a11yTitle="Session Menu"
                className={styles.rightMenu}
              >
                <Anchor href="/user/profile">
                  My Profile
                </Anchor>
                <Anchor href="/logout" onClick={onLogout}>
                  Logout
                </Anchor>
              </Menu>
            );
          case 'admin':
            return (
              <Menu
                tabIndex={tabIndex}
                icon={<SessionIcon user={user} />}
                dropAlign={{ bottom: 'bottom' }}
                a11yTitle="Session"
                inline={false}
                className={styles.rightMenu}
              >
                <Anchor
                  className={pathname === '/user/profile' ? 'navLink active' : 'navLink'}
                  href="/user/profile"
                >
                  My Profile
                </Anchor>
                <Anchor
                  className={pathname === '/admin/dashboard' ? 'navLink active' : 'navLink'}
                  href="/admin/dashboard"
                >
                  Dashboard
                </Anchor>
                <Anchor href="/logout" onClick={onLogout}>
                  Logout
                </Anchor>
              </Menu>
            );
          default:
            return (
              <NoUserMenu pathname={pathname} />
            );
        }
      } else {
        return (
          <NoUserMenu pathname={pathname} />
        );
      }
    })()}
  </span>
);

SessionMenu.propTypes = {
  user: PropTypes.object,
  pathname: PropTypes.string.isRequired,
  tabIndex: PropTypes.number.isRequired,
};

SessionMenu.defaultProps = {
  tabIndex: 0,
};

export default cssModules(SessionMenu, styles);
