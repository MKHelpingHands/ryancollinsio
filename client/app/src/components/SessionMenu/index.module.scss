
.userAvatar {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  border: 2px solid transparent;
  overflow: hidden;
  cursor: pointer;
  transition: all .3s ease-in-out;
}

.profileName {
  margin-top: 12px;
  margin-left: 5px;
  margin-right: 5px;
}

.rightMenu {
  margin-right: 20px !important;
}
