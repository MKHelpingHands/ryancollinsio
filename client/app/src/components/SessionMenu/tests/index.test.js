import SessionMenu from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<SessionMenu />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SessionMenu />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
