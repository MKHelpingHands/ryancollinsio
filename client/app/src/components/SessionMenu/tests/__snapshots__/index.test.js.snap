exports[`<SessionMenu /> should render with default props 1`] = `
<span>
  <NoUserMenu />
</span>
`;
